runtime/onert/core/src/exporter/CircleExporter.cc

```diff
@@ -81,6 +83,50 @@ void CircleExporter::updateWeight(const std::unique_ptr<exec::Execution> &exec)
     });
 }
 
+void CircleExporter::updateMetadata(const std::unique_ptr<ir::train::TrainingInfo> &training_info)
+{
+  const char *const TRAININFO_METADATA_NAME = "CIRCLE_TRAINING";
+
+  TrainInfoBuilder tbuilder(training_info);
+  static std::mutex mutex;
+
+  bool found = false;
+  for (const auto &meta : _model->metadata)
+  {
+    if (meta->name == std::string{TRAININFO_METADATA_NAME})
+    {
+      const uint32_t buf_idx = meta->buffer;
+      auto &buffer = _model->buffers.at(buf_idx);
+
+      if (tbuilder.size() != buffer->data.size())
+      {
+        buffer->data.resize(tbuilder.size());
+        buffer->size = tbuilder.size();
+      }
+
+      memcpy(&buffer->data[0], tbuilder.get(), tbuilder.size());
```

Comment:
How about using `buffer->data.data()` instead of `&buffer->data[0]` for readability?
```suggestion
      memcpy(buffer->data.data(), tbuilder.get(), tbuilder.size());
```

Reply:
Here is the answer about it by "code.i":

---

Could you explain if can we use `buffer->data.data()` instead of `&buffer->data[0]`?

---

Yes, you can use buffer->data.data() instead of &buffer->data[0] in the memcpy() function call. The std::vector<uint8_t>::data() function returns the address of the first element of the std::vector<uint8_t>. Therefore, you can safely use buffer->data.data() instead of &buffer->data[0] when calling the memcpy() function.